src/services/validation.js

```diff
@@ -1,36 +1,41 @@
 import firebaseAPI from "@/services/API/checkUser.js";
 import { sendPlio } from "@/services/sendPlio";
+import {sendSQSMessage} from "@/services/API/sqs";
 
-//this function is invoked only for the check of SRN's.
-export async function validateSRN(userID, validateCount, isSingleEntryOnly, redirectID, doesUserExist){
+// this function is invoked only for the check of SRN's.
+// validateCount = 0, when the user gets validated on the first try
+// validateCount = 1, when the user fails the first validation
 
+export async function validateSRN(userID, validateCount, isSingleEntryOnly, redirectID, doesUserExist, purpose, purposeParams, redirectTo){
+    var authType="SRN"
     var invalidLoginMessage = ""
-    //checks the basic conditions of SRN: 
-    //      - starts with only '1'
+
+    //checks the basic condition of SRN: 
     //      - does not contain same number patterns like '111...', '222...', etc.
-    if(!(/^([0-9])\1*$/.test(userID)) && userID.toString()[0] == '1'){
+    if(!(/^([0-9])\1*$/.test(userID))){
 
         //if the basic conditions are satisified, only then the request is sent for further checking
         doesUserExist = await firebaseAPI.checkUserExists(userID);
     }
 
     // this condition checks if the user is getting authenticated the first time. Just shows an error message.
     if (!doesUserExist && validateCount == 0) {
-    validateCount = 1;
-    invalidLoginMessage = "Please enter correct SRN / कृपया सही SRN दर्ज करें";
+        validateCount = 1;
+        invalidLoginMessage = "Please enter correct SRN / कृपया सही SRN दर्ज करें";
     }
-    //this condition checks the second time, since still not valid, just changes the flag and continues with the plio.
-    else if (!doesUserExist && validateCount == 1) {
-    validateCount = 2;
-    sendPlio(isSingleEntryOnly, userID, redirectID);
-
-    } else {
-    sendPlio(isSingleEntryOnly, userID, redirectID);
 
+    //this condition is entered in one of the three conditions:
+    //  - either the user is found the first time -> doesUserExist = true, validateCount = 0
+    //  - either the user is found the second time -> doesUserExist = true, validateCount = 1
+    //  - either the user is not found the second time -> doesUserExist = false, validateCount = 1
```

Comment:
If `validateCount` is 1 in both cases, why are the conditions separate? If the final message sent depends only on `doesUserExist`, why even have the `validateCount` variable?

Reply:
It doesn't totally depend on `doesUserExist`. Think about it like this:
- When `validateCount` = 0, if `doesUserExist` = true, validated on first try.
                                                  if `doesUserExist` = false, `validateCount` = 1, and validation is tried again.
 What to do during a validation cycle depends on what the value of `validateCount` is. `doesUserExist` helps us in deciding if we want to validate again and for the flag in BQ.

Reply:
Don't see the logic here. You have only one `if`, so you're checking only one thing, no? 

IMO (maybe I'm wrong) only one variable can do the work of both:

`validateCount = -1 => user not validated`
`validateCount = 0 => user does not exist`
`validateCount = 1 => user does not exist, SRN validated`
`validateCount = 2 => user exists, SRN validated`

`validateCount` is also a weird name for this. Maybe call it `validationState`? That's why I originally said use an enum. Will make the code much clearer to read because then you can use variable names instead of numbers. But numbers is fine (and more efficient in JS for what it's worth)